File: backend/battleship/src/main/java/com/sbz/battleship/domain/model/Game.java
package com.sbz.battleship.domain.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;
import java.util.UUID;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Document(collection = "games")
public class Game {

    @Id
    private UUID id;
    private UUID playerId;

    private List<Move> playerMoves;
    private Ships computerShips;

    private List<Move> computerMoves;
    private Ships playerShips;

    private Boolean winner; // true computer, false player

}

File: frontend/src/app/shared/model/game.ts
import { Player } from './player';
import { Ship } from './ship';
import { Move } from './move';
import { Ships } from './ships';


export class Game {

    id: string;
    player: Player;

    playerMoves: Move[]; 
    computerShips: Ships;

    computerMoves: Move[]; 
    playerShips: Ships;

    winner: boolean;

    constructor() {
        this.playerMoves = []; 
        this.computerMoves = []; 
    }
}
